serving/src/app/callback/store/redisstore.rs

```diff
@@ -264,4 +271,47 @@ mod tests {
         let result = redis_conn.retrieve_datum("non_existent_key").await;
         assert!(matches!(result, Err(Error::StoreRead(_))));
     }
+
+    #[tokio::test]
+    async fn test_redis_ttl() {
+        let redis_connection = RedisConnection::new("redis://127.0.0.1:6379", 10)
+            .await
+            .expect("Failed to connect to Redis");
+
+        let key = uuid::Uuid::new_v4().to_string();
+        let value = Arc::new(CallbackRequest {
+            id: String::from("test-redis-ttl"),
+            vertex: String::from("vertex"),
+            cb_time: 1234,
+            from_vertex: String::from("next_vertex"),
+            tags: None,
+        });
+
+        // Save with TTL of 1 second
+        let mut conn_manager = redis_connection.conn_manager.clone();
+        RedisConnection::write_to_redis(
+            &mut conn_manager,
+            &key,
+            &serde_json::to_vec(&*value).unwrap(),
+        )
+        .await
+        .expect("Failed to write to Redis");
+
+        // Immediately check existence
+        let exists: bool = conn_manager
+            .exists(&key)
+            .await
+            .expect("Failed to check existence immediately");
+        assert!(exists, "Key should exist immediately after saving");
```

Comment:
use `TTL` command to see whether the value has been set. do not sleep. check for existence or TTL being set.